transport/peer/x/peerlist/roundrobin/peerring.go

```diff
@@ -38,18 +37,14 @@ func NewPeerRing(capacity int) *PeerRing {
 // PeerRing provides a safe way to interact (Add/Remove/Get) with a potentially
 // changing list of peer objects
 type PeerRing struct {
-	lock sync.Mutex
-
 	peerToNode map[string]*peerRingNode
 	nextNode   *peerRingNode
 }
 
 // Add a transport.Peer to the end of the PeerRing, if the ring is empty
 // it initializes the nextNode marker
+// Must be run inside a mutex.Lock()
```

Comment:
That's a weird requirement for a top-level method of a type. There's no
`mutex` in the context of PeerRing.

Instead, remove any references to locks in the comments for PeerRing and
mention in the docstring for PeerRing that this is not thread safe. That
explains to the caller that locking is their concern.